src/rgw/rgw_main.cc

```diff
@@ -537,6 +537,8 @@ int main(int argc, const char **argv)
 
   derr << "shutting down" << dendl;
 
+  delete reloader;
+
```

Comment:
can you help me understand how deleting the reloader early addresses the segfault? also note that `realm_watcher` will still hold a reference to this `reloader`

Reply:
Oh, I only tested restart after reload. I think add "realm_watcher.watch_stop()" will fix reload after restart situation, and local test passed. whats your opinion?